casper/src/main/scala/coop/rchain/casper/rholang/RuntimeManager.scala

```diff
@@ -39,25 +40,24 @@ import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.FiniteDuration
 
 trait RuntimeManager[F[_]] {
-  def captureResults(
-      startHash: StateHash,
-      deploy: Signed[DeployData]
-  ): F[Seq[Par]]
   def replayComputeState(startHash: StateHash)(
       terms: Seq[ProcessedDeploy],
-      systemDeploys: Seq[ProcessedSystemDeploy],
+      rand: Blake2b512Random,
       blockData: BlockData,
-      withCostAccounting: Boolean
+      withCostAccounting: Boolean,
+      systemDeploys: Seq[ProcessedSystemDeploy]
```

Comment:
Why did you move system deploys? They are executed in a similar way as user deploys and use random generator.